### PR TITLE
Implement miner_setMinPriorityFee and miner_getMinPriorityFee

### PR DESCRIPTION
**Pull request by Gabriel-Trintinalia**
## PR description
Built on top of https://github.com/fab-10/besu/tree/min-priority-fee, which adds `--min-priority-fee` CLI option

Adds two new JsonRPC methods: 

- miner_setMinPriorityFee - Sets the value of `miningParameters.minPriorityFeePerGas`
- miner_getMinPriorityFee - Returns the current value of `miningParameters.minPriorityFeePerGas`

This PR does not add the transactionSelector based on `minPriorityFeePerGas`. It will be added in a different PR

### CODE REVIEW

File: besu/src/main/java/org/hyperledger/besu/cli/options/MiningOptions.java [hunk 1]
```diff
@@ -93,10 +94,19 @@ public class MiningOptions implements CLIOptions<MiningParameters> {
       arity = "1")
   private Wei minTransactionGasPrice = DEFAULT_MIN_TRANSACTION_GAS_PRICE;
 
+  @Option(
+      names = {"--min-priority-fee"},
+      description =
+          "Minimum priority fee per gas (in Wei) offered by a transaction for it to be included in a mined "
```
Comment (macfarla) [inline besu/src/main/java/org/hyperledger/besu/cli/options/MiningOptions.java:100]:
```suggestion
          "Minimum priority fee per gas (in Wei) offered by a transaction for it to be included in a "
```

File: besu/src/main/java/org/hyperledger/besu/cli/options/MiningOptions.java [hunk 2]
```diff
@@ -93,10 +94,19 @@ public class MiningOptions implements CLIOptions<MiningParameters> {
...
+              + "block (default: ${DEFAULT-VALUE})",
+      arity = "1")
+  private Wei minPriorityFeePerGas = DEFAULT_MIN_PRIORITY_FEE_PER_GAS;
+
   @Option(
       names = {"--target-gas-limit"},
       description =
-          "Sets target gas limit per block. If set, each block's gas limit will approach this setting over time if the current gas limit is different.")
+          "Sets target gas limit per block."
+              + " If set, each block's gas limit will approach this setting over time if the current gas limit is different.")
```
Comment (macfarla) [inline besu/src/main/java/org/hyperledger/besu/cli/options/MiningOptions.java:109]:
```suggestion
              + " If set, each block's gas limit will approach this setting over time.")
```

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/miner/MinerSetMinPriorityFee.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  @Override
+  public String getName() {
+    return RpcMethod.MINER_SET_MIN_PRIORITY_FEE.getMethodName();
+  }
+
+  @Override
+  public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
+    try {
+      final Wei minPriorityFee = Wei.of(requestContext.getRequiredParameter(0, Long.class));
+      miningParameters.setMinPriorityFeePerGas(minPriorityFee);
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/miner/MinerSetMinPriorityFee.java:46]: nit - consistency between minPriorityFee and minPriorityFeePerGas

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/MiningParameters.java [hunk 4]
```diff
@@ -65,6 +65,15 @@ public MiningParameters setMinTransactionGasPrice(final Wei minTransactionGasPri
     return this;
   }
 
+  public Wei getMinPriorityFeePerGas() {
+    return getMutableRuntimeValues().minPriorityFeePerGas;
+  }
+
+  public MiningParameters setMinPriorityFeePerGas(final Wei minPriorityFeePerGas) {
+    getMutableRuntimeValues().minPriorityFeePerGas = minPriorityFeePerGas;
```
Comment (macfarla) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/MiningParameters.java:73]: do we have to worry about ensuring the minPriorityFee stays < maxPriorityFee

**Review by pinges — APPROVED** [review]
LGTM

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/miner/MinerSetMinPriorityFee.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+  @Override
+  public String getName() {
+    return RpcMethod.MINER_SET_MIN_PRIORITY_FEE.getMethodName();
+  }
+
+  @Override
+  public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
+    try {
+      final Wei minPriorityFeePerGas = Wei.of(requestContext.getRequiredParameter(0, Long.class));
```
Comment (Filter94) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/miner/MinerSetMinPriorityFee.java:45]: There's a slight concern on the signed Long usage for Wei, which is much smaller than `UInt256`, which is what under the hood of Wei can is a theoretical maximum.

**Pull request merged**